File: user-srv-v2/internal/domain/device.go
package domain

import (
	"github.com/google/uuid"
)

type Device struct {
	Id                     uuid.UUID              `gorm:"column:id;type:varchar(36);primaryKey" json:"id"`
	DeviceId               uuid.UUID              `gorm:"column:device_id;type:varchar(36)" json:"deviceId"`
	DeviceToken            string                 `gorm:"column:device_token" json:"deviceToken"`
	DeviceIdentifier       string                 `gorm:"column:device_identifier" json:"deviceIdentifier"`
	Language               string                 `gorm:"column:language" json:"language"`
	Platform               string                 `gorm:"column:platform" json:"platform"`
	IsActive               bool                   `gorm:"column:is_active" json:"isActive"`
	SubscriptionIdentifier string                 `gorm:"column:subscription_identifier" json:"subscriptionIdentifier"`
	UserInfoId             *uuid.UUID             `gorm:"column:user_info_id;type:varchar(36)" json:"userInfoId"`
	NotificationPreference NotificationPreference `gorm:"foreignKey:DeviceForeignKey" json:"notificationPreference"`
}

File: user-srv/src/Pi.User/Pi.User.Domain/AggregatesModel/UserInfoAggregate/Device.cs
using Pi.Common.SeedWork;
using Pi.User.Domain.SeedWork;

namespace Pi.User.Domain.AggregatesModel.UserInfoAggregate;

public class Device : Entity, IAuditableEntity
{
    public Device(Guid deviceId, string deviceToken, string language, string deviceIdentifier, string platform, string subscriptionIdentifier)
    {
        DeviceId = deviceId;
        DeviceToken = deviceToken;
        DeviceIdentifier = deviceIdentifier;
        Language = language;
        Platform = platform;
        IsActive = true;
        SubscriptionIdentifier = subscriptionIdentifier;
        NotificationPreference = new NotificationPreference(true, true, true, true, true);
    }

    public Guid Id { get; private set; }
    public Guid DeviceId { get; private set; }
    public string DeviceToken { get; private set; }
    public string DeviceIdentifier { get; private set; }
    public string Language { get; private set; }
    public string Platform { get; private set; }
    public bool IsActive { get; private set; }
    public string SubscriptionIdentifier { get; private set; }
    public NotificationPreference NotificationPreference { get; }
    public Guid? UserInfoId { get; }
    public UserInfo? UserInfo { get; }

    public Device UpdateDeviceToken(string deviceToken)
    {
        DeviceToken = deviceToken;

        return this;
    }

    public Device UpdateLanguage(string language)
    {
        Language = language;

        return this;
    }

    public Device UpdateDeviceIdentifier(string newDeviceIdentifier)
    {
        DeviceIdentifier = newDeviceIdentifier;
        return this;
    }

    public Device UpdateDevicePlatform(string platform)
    {
        Platform = platform;
        return this;
    }

    public Device MarkInactive()
    {
        IsActive = false;

        return this;
    }

    public Device UpdateSubscription(string subscriptionIdentifier)
    {
        SubscriptionIdentifier = subscriptionIdentifier;

        return this;
    }

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}